### PR TITLE
Testclusters: conver remaining x-pack 

### PR DESCRIPTION
**Pull request by alpar-t**
With this PR all of the `x-pack` folder is converted to use testclusters.

### CODE REVIEW

File: x-pack/plugin/build.gradle [hunk 1]
```diff
@@ -25,14 +26,13 @@ subprojects {
...
+      File successMarker = file("$buildDir/markers/featureAware")
+      task featureAwareCheck(type: LoggedExec) {
         description = "Runs FeatureAwareCheck on main classes."
         dependsOn project.configurations.featureAwarePlugin
-
-        final File successMarker = new File(project.buildDir, 'markers/featureAware')
         outputs.file(successMarker)
 
-        executable = new File(project.runtimeJavaHome, 'bin/java')
+        executable = "${project.runtimeJavaHome}/bin/java"
```
Comment (mark-vieira) [inline x-pack/plugin/build.gradle]: Let's avoid doing this. I've already removed a bunch of instances of this while performing other refactorings. There is no reason to do these build time checks with a forked JVM. The code being executed here is build logic, not application logic, and therefore there's no benefit to using the runtime JVM yet we pay the cost of forking another JVM. We don't run the compiler, or checkstyle with the runtime Java home, so we shouldn't here either.

File: x-pack/plugin/build.gradle [hunk 2]
```diff
@@ -125,8 +105,29 @@ task copyKeyCerts(type: Copy) {
...
+  /*
+   * We have to disable setting the number of available processors as tests in the same JVM randomize processors and will step on each
+   * other if we allow them to set the number of available processors as it's set-once in Netty.
+   */
+  systemProperty 'es.set.netty.runtime.available.processors', 'false'
+
+
+  // TODO: fix this rest test to not depend on a hardcoded port!
+  def blacklist = ['getting_started/10_monitor_cluster_health/*']
+  boolean snapshot = "true".equals(System.getProperty("build.snapshot", "true"))
```
Comment (mark-vieira) [inline x-pack/plugin/build.gradle]: This just looks odd to me. Let's just use `Boolean.valueOf()` as we do in other places.

File: x-pack/plugin/sql/qa/build.gradle [hunk 3]
```diff
@@ -96,20 +94,20 @@ subprojects {
...
     }
 
-    task runqa(type: RunTask) {
-      setting 'xpack.monitoring.enabled', 'false'
-      setting 'xpack.ml.enabled', 'false'
-      setting 'xpack.watcher.enabled', 'false'
-      setting 'script.max_compilations_rate', '1000/1m'
+    task runqa {
+      doFirst {
+        println "Run with `-Dtestclusters.inspect.failure=true integTest` to leave the cluster running after failure"
```
Comment (mark-vieira) [inline x-pack/plugin/sql/qa/build.gradle]: Is this just instructing folks to use the `run` task instead? If so should we be more specific? Also, if we are effectively "deprecating" this task, we should do so in a `taskGraphReady` callback so that we catch this scenario earlier for the user.

File: x-pack/plugin/sql/qa/security/build.gradle [hunk 4]
```diff
@@ -29,38 +29,23 @@ subprojects {
...
-  integTestCluster {
+  testClusters.integTest {
+    distribution = "DEFAULT"
     // Setup auditing so we can use it in some tests
     setting 'xpack.security.audit.enabled', 'true'
     setting 'xpack.security.enabled', 'true'
     setting 'xpack.license.self_generated.type', 'trial'
     // Setup roles used by tests
-    extraConfigFile 'roles.yml', '../roles.yml'
+    extraConfigFile 'roles.yml', file('../roles.yml')
```
Comment (mark-vieira) [inline x-pack/plugin/sql/qa/security/build.gradle]: I think this should just be `file('roles.yml')` since that file exists in the root of the project directory (peer of the build script). I suspect the relative path was needed before due to this being a simple string and how the test cluster working directory was set.

File: x-pack/qa/reindex-tests-with-security/build.gradle [hunk 5]
```diff
@@ -22,10 +21,11 @@ forbiddenPatterns {
 
 File caFile = project.file('src/test/resources/ssl/ca.p12')
 
-integTestCluster {
+testClusters.integTest {
+  distribution = "DEFAULT"
   // Whitelist reindexing from the local node so we can test it.
-  extraConfigFile 'http.key', project.projectDir.toPath().resolve('src/test/resources/ssl/http.key')
```
Comment (mark-vieira) [inline x-pack/qa/reindex-tests-with-security/build.gradle]: 👍  Thank you for this! Not sure why the usages of `Path` are littered in our build scripts but simply using `Project.file()` is so much more clear and concise.

File: x-pack/qa/smoke-test-plugins-ssl/build.gradle [hunk 6]
```diff
@@ -74,18 +70,13 @@ integTestCluster {
...
+  user username: "monitoring_agent", password: "x-pack-test-password", role: "remote_monitoring_agent"
 
-  waitCondition = { NodeInfo node, AntBuilder ant ->
-    WaitForHttpResource http = new WaitForHttpResource("https", node.httpUri(), numNodes)
-    http.setTrustStoreFile(clientKeyStore)
-    http.setTrustStorePassword("testclient")
-    http.setUsername("test_user")
-    http.setPassword("x-pack-test-password")
-    return http.wait(5000)
+  project(':plugins').getChildProjects().each { pluginName, pluginProject ->
```
Comment (mark-vieira) [inline x-pack/qa/smoke-test-plugins-ssl/build.gradle]: Why is this necessary if we are using the default distribution?

File: x-pack/qa/third-party/jira/build.gradle [hunk 7]
```diff
@@ -11,23 +12,18 @@ dependencies {
...
 
-ext {
-  jiraUrl      = System.getenv('jira_url')
-  jiraUser     = System.getenv('jira_user')
-  jiraPassword = System.getenv('jira_password')
-  jiraProject  = System.getenv('jira_project')
-}
 
-integTestCluster {
+String jiraUrl      = System.getenv('jira_url')
```
Comment (mark-vieira) [inline x-pack/qa/third-party/jira/build.gradle]: Can we ditch the odd whitespace here?

**Review by mark-vieira — APPROVED** [review]
Some minor comments but otherwise LGTM. I can see the light! 😄

File: x-pack/qa/smoke-test-plugins-ssl/build.gradle (hunk 6 above)
Comment (atorok) [reply x-pack/qa/smoke-test-plugins-ssl/build.gradle]: The default distribution has all the modules and the x-pack plugins, but  doesn't have the OSS plugins  ( there's a handful of those in `:plugins` ) for different specialized features, so the test essentially verifies that these can be installed.

File: x-pack/plugin/build.gradle (hunk 1 above)
Comment (atorok) [reply x-pack/plugin/build.gradle]: I completely agree but don't want to pick that up right now, there were so many things I saw that should be cleaned up, but wanted to keep focus on testclusters.

File: x-pack/plugin/sql/qa/build.gradle (hunk 3 above)
Comment (atorok) [reply x-pack/plugin/sql/qa/build.gradle]: These projects had something similar to a run that stared up  cluster identical to the testcluster. We have no equivalent of `run` yet and that still uses the old method. 
I suspect the intent was to ease debugging , and that's exactly what the command `-Dtestclusters.inspect.failure=true integTest` does, it runs tests and leaves the cluster running if it fails. I don't think it's worth looking at the task graph as I expect the task to be run directly for a single project, so doing it via the task graph would not yield a significant change  with that use-case.

File: x-pack/plugin/sql/qa/security/build.gradle (hunk 4 above)
Comment (atorok) [reply x-pack/plugin/sql/qa/security/build.gradle]: that should have failed with a missing file exception as we do check for existence. 
We are in a `subprojects` block so `file` resolves to the subproject and thus needs to go to the parent. I;m rewriting this to make it more obvious.

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: x-pack/plugin/build.gradle (hunk 1 above)
Comment (mark-vieira) [reply x-pack/plugin/build.gradle]: Well we are talking about just removing this line not any kind of significant refactor. Let's just go ahead and make this change while we're here.

Comment (atorok) [reply x-pack/plugin/build.gradle]: Not quite, it's a exec not a java exec, so there's a bit more to it.

Comment (mark-vieira) [reply x-pack/plugin/build.gradle]: Ah, I see.
